test/converters/xcresult3/model3/conversion.go

```diff
@@ -0,0 +1,98 @@
+package model3
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+func Convert(data *TestData) (*TestSummary, error) {
+	summary := TestSummary{}
+
+	for _, testPlanNode := range data.TestNodes {
+		if testPlanNode.Type != TestNodeTypeTestPlan {
+			return nil, fmt.Errorf("test plan expected but got: %s", testPlanNode.Type)
+		}
+
+		testPlan := TestPlan{Name: testPlanNode.Name}
+
+		for _, testBundleNode := range testPlanNode.Children {
+			if testBundleNode.Type != TestNodeTypeUnitTestBundle && testBundleNode.Type != TestNodeTypeUITestBundle {
+				return nil, fmt.Errorf("test bundle expected but got: %s", testBundleNode.Type)
+			}
+
+			testBundle := TestBundle{Name: testBundleNode.Name}
+
+			for _, testSuiteNode := range testBundleNode.Children {
+				if testSuiteNode.Type != TestNodeTypeTestSuite {
+					return nil, fmt.Errorf("test suite expected but got: %s", testSuiteNode.Type)
+				}
+
+				testSuite := TestSuite{Name: testSuiteNode.Name}
+
+				for _, testCaseNode := range testSuiteNode.Children {
+					if testCaseNode.Type != TestNodeTypeTestCase {
+						return nil, fmt.Errorf("test case expected but got: %s", testCaseNode.Type)
```

Comment:
These errors made me wonder if we should fall back to the legacy parser in case of an error in the new parser, what do you think of this idea?

Reply:
Great idea and I made this change.